Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,240 @@ public static class BalanceStats
+                bigLoss = loss;
+            }
+
+            return bigLoss;
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+                
+                if (isBalanceValid && balance >= maxBalance)
+                {
+                    if (balance > maxBalance)
+                    {
+                        name = new StringBuilder();
+                    }
+                    if (name.Length != 0)
+                    {
+                        name.Append(", ");
+                    }
+                    name.Append(peopleBalances[0]);
+
+                    maxBalance = balance;
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            var word1 = name.ToString().Contains(", ") ? "are" : "is";
+            var word2 = name.ToString().Contains(", ") ? "people" : "person";
+            return $"{ReplaceLastComma(name.ToString())} {word1} the richest {word2}. {maxBalanceNoComma}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var minBalance = decimal.MaxValue;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var isBalanceValid = decimal.TryParse(peopleBalances[peopleBalances.Length - 1], out var balance);
+
+                if (isBalanceValid && balance <= minBalance)
+                {
+                    if (balance < minBalance)
+                    {
+                        name = new StringBuilder();
+                    }
+                    if (name.Length != 0)
+                    {
+                        name.Append(", ");
+                    }
+                    name.Append(peopleBalances[0]);
+
+                    minBalance = balance;
+                }
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
```

Comment:
Split the bottom part- format result

Reply:
Not sure what this was meant but I deleted useless IF statement code when minBalance less than 0